src/main/java/duke/util/TaskList.java

```diff
@@ -0,0 +1,205 @@
+    private Ui ui = new Ui();
+
+    /**
+     * Creates a new instance of tasklist that creates a new list of tasks.
+     */
+    public TaskList() {
+        taskList = new ArrayList<>();
+    }
+
+    /**
+     * Creates a new instance of tasklist that loads the saved list of tasks.
+     */
+    public TaskList(Scanner sc) throws FileNotFoundException, DukeException {
+        this.taskList = new ArrayList<>();
+        readFromFile(sc);
+    }
+
+    /**
+     * Loads the saved list of tasks
+     */
+    private void readFromFile(Scanner sc) throws DukeException {
+        while (sc.hasNext()) {
+            String[] savedData = sc.nextLine().split(" \\| ");
+            String taskType = savedData[0];
+            Task task;
+            taskCount++;
+            switch (taskType) {
+            case "T":
+                task = new Todo(savedData[2]);
+                break;
+            case "E":
+                if (savedData.length == 3) {
+                    task = new Event(savedData[2], savedData[3]);
+                } else { // If time is included for the task
+                    task = new Event(savedData[2], savedData[3] + " " + savedData[4]);
+                }
+                break;
+            case "D":
+                if (savedData.length == 3) {
+                    task = new Deadline(savedData[2], savedData[3]);
+                } else { // If time is included for the task
+                    task = new Deadline(savedData[2], savedData[3] + " " + savedData[4]);
+                }
+                break;
+            default:
+                task = new Todo("");
+            }
+            if (savedData[1].equals("1")) {
+                task.markAsDone();
+            }
+            taskList.add(task);
+        }
+    }
+
+    /**
+     * Adds a task into list of tasks.
+     *
+     * @param task List of tasks.
+     */
+    public String add(Task task) {
+        StringBuilder str = new StringBuilder();
+        if (task != null) {
+            this.taskList.add(task);
+            this.taskCount++;
+            str.append(TASK_ADDED + taskList.get(this.taskCount - 1).toString());
+            str.append("\nNow you have " + this.taskCount
+                    + " tasks in your task list arrr, better get workin' aye!\n");
+            str.append(ui.requestNextCommand());
+            return str.toString();
+        } else {
+            str.append(ui.showError("\tTask is invalid matey :-(, please try again!\n"));
+            return str.toString();
+        }
+    }
+
+    /**
+     * Deletes a task from list of tasks.
+     *
+     * @param taskIndex 0-based index task number to be deleted.
+     */
+    public String delete(int taskIndex) {
+        StringBuilder str = new StringBuilder();
+        try {
+            Task task = taskList.get(taskIndex);
+            this.taskList.remove(taskIndex);
+            this.taskCount--;
+            str.append(ui.deleteTask() + task+ "\n" + ui.requestNextCommand());
```

Comment:
There is a missing white space here.
```suggestion
            str.append(ui.deleteTask() + task + "\n" + ui.requestNextCommand());
```